magma/backend/mlir/mlir_compiler.py

```diff
@@ -45,7 +45,11 @@ def run_pre_uniquification_passes(self):
         # could introduce more conditional logic, or they could
         # trigger elaboration on values used in existing coditiona
         # logic (which modifies the when builder).
-        run_when_passes(self.main, self.opts.get("emit_when_assertions", False))
+        run_when_passes(
+            self.main,
```

Comment:
btw, we should just use `self._compile_to_mlir_opts.{flatten_all_tuples,emit_when_assertions}` here. same above (L42). Can do this in a separate PR if desired.